src/datatypes2/src/vectors/string.rs

```diff
@@ -0,0 +1,369 @@
+        assert_eq!(Value::String("world".into()), vector.get(2));
+
+        let mut iter = vector.iter_data();
+        assert_eq!("hello", iter.next().unwrap().unwrap());
+        assert_eq!(None, iter.next().unwrap());
+        assert_eq!("world", iter.next().unwrap().unwrap());
+        assert_eq!(None, iter.next());
+    }
+
+    #[test]
+    fn test_string_vector_builder() {
+        let mut builder = StringVectorBuilder::with_capacity(3);
+        builder.push_value_ref(ValueRef::String("hello")).unwrap();
+        assert!(builder.push_value_ref(ValueRef::Int32(123)).is_err());
+
+        let input = StringVector::from_slice(&["world", "one", "two"]);
+        builder.extend_slice_of(&input, 1, 2).unwrap();
+        assert!(builder
+            .extend_slice_of(&crate::vectors::Int32Vector::from_slice(&[13]), 0, 1)
+            .is_err());
+        let vector = builder.to_vector();
+
+        let expect: VectorRef = Arc::new(StringVector::from_slice(&["hello", "one", "two"]));
+        assert_eq!(expect, vector);
+    }
+
+    #[test]
+    fn test_string_vector_misc() {
+        let strs = vec!["hello", "greptime", "rust"];
+        let v = StringVector::from(strs.clone());
+        assert_eq!(3, v.len());
+        assert_eq!("StringVector", v.vector_type_name());
+        assert!(!v.is_const());
+        assert_eq!(Validity::AllValid, v.validity());
+        assert!(!v.only_null());
+        assert_eq!(128, v.memory_size());
+
+        for (i, s) in strs.iter().enumerate() {
+            assert_eq!(Value::from(*s), v.get(i));
+            assert_eq!(ValueRef::from(*s), v.get_ref(i));
+            assert_eq!(Value::from(*s), v.try_get(i).unwrap());
+        }
+
+        let arrow_arr = v.to_arrow_array();
+        assert_eq!(3, arrow_arr.len());
+        assert_eq!(&DataType::Utf8, arrow_arr.data_type());
+    }
+
+    #[test]
+    fn test_serialize_string_vector() {
+        let mut builder = StringVectorBuilder::with_capacity(3);
+        builder.push(Some("hello"));
+        builder.push(None);
+        builder.push(Some("world"));
+        let string_vector = builder.finish();
+        let serialized =
+            serde_json::to_string(&string_vector.serialize_to_json().unwrap()).unwrap();
+        assert_eq!(r#"["hello",null,"world"]"#, serialized);
+    }
+
+    #[test]
+    fn test_from_arrow_array() {
+        let mut builder = MutableStringArray::new();
+        builder.append_option(Some("A"));
+        builder.append_option(Some("B"));
+        builder.append_null();
+        builder.append_option(Some("D"));
+        let string_array: StringArray = builder.finish();
+        let vector = StringVector::from(string_array);
+        assert_eq!(
+            r#"["A","B",null,"D"]"#,
+            serde_json::to_string(&vector.serialize_to_json().unwrap()).unwrap(),
+        );
+    }
+
+    #[test]
+    fn test_from_non_option_string() {
+        let corpus = vec!["😅😅😅", "😍😍😍😍", "🥵🥵"];
```

Comment:
You can also test if StringVector can correctly handle [NUL](https://www.compart.com/en/unicode/U+0000)  😅😅😅